catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/ws/SecureWebSocketServlet.java

```diff
@@ -90,15 +108,19 @@ public void onError(Session session, Throwable ex) {
 
     @OnWebSocketMessage
     public void onMessage(Session session, String message) {
-      runWithUser(
-          session,
-          () -> {
-            try {
-              ws.onMessage(session, message);
-            } catch (IOException e) {
-              LOGGER.error("Failed to receive ws message.", e);
-            }
-          });
+      if (securityTokenHolder.getRealmTokenMap().size() == 0) {
```

Comment:
`securityTokenHolder.getRealmTokenMap().size() == 0` is used in two places.
How about extracting it into a predicate method like `boolean isMissingTokens(ecurityTokenHolder securityTokenHolder)`

Reply:
I can't just throw the exception in this method; I need to call the `onError` method with the exception as an argument. I could catch the exception and call `onError`, but then I'm back to the same cyclomatic complexity. I will make a helper method.

Reply:
Oh. Duh. I misread that. :)